stockflux-securities-master/src/components/TextField.js

```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import "./TextField.css";
+import PropTypes from "prop-types";
+
+const TextField = ({ onChange, id, value, readOnly, className, ...props }) => {
+  return (
+    <input
+      {...props}
+      onChange={onChange}
+      id={id}
+      value={value}
+      readOnly={readOnly}
+      className={`input-form-input${readOnly ? " read-only" : ""}${
+        className ? ` ${className}` : ""
+      }`}
+    />
+  );
+};
+
+TextField.propTypes = {
+  onChange: PropTypes.func.isRequired,
```

Comment:
I'm not sure that any of these should be required. I think it's totally valid to be able to omit an `onChange` function and an `id`. A `value` is only really necessary if an existing value is actually present. In most cases they will be provided but I don't think they should be made mandatory. It might actually be worth removing these 3 fields from the destructured arguments to the component and just allowing the `{...props}` spread to apply them on the input element. The prop types you've defined should take care of ensuring that the correct types are given when the component is used.

Reply:
yep done